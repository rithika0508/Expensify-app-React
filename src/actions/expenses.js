import { v1 as uuidv1 } from 'uuid';

//add expense
export const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => {
    return {
        type: 'ADD-EXPENSE',
        expenses: {
            id: uuidv1(),
            description,
            note,
            amount,
            createdAt
        }
    }
}

// delete expense
export const deleteExpense = ({ id } = {}) => {
    return {
        type: 'DELETE-EXPENSE',
        id,
    };
}

// edit expense

export const editExpense = (id, updates) => {
    
    return {
        type: 'EDIT-EXPENSE',
        id,
        updates
    }
    
}