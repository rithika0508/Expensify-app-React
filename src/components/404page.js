import React from 'react';
const Notfoundpage = () => (
    <div>404 Page is not available
    </div>
);
export default Notfoundpage;