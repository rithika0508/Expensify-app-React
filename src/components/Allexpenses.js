import React from 'react';
import { connect } from 'react-redux';
import SingleExpense from './SingleExpense';
import visibleexpenses from '../selectors/expenses'


export const AllExpenses = (props) => {
    return (
        <div>
            {props.expenses.length==0 && <p>Add an expense to get started!</p>}
            {props.expenses.map((expense) => <SingleExpense key={expense.id} {...expense}/>)}
        </div>
    );
}
const mapStateToProps = (state) => {
    return {
        expenses: visibleexpenses(state.expenses, state.filter)
    }
}
const Connected = connect(mapStateToProps)(AllExpenses);
export default Connected;