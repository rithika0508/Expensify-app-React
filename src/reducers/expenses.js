
const expensesReducersDefaultState = []
const expensesReducers = (state = expensesReducersDefaultState, action) => {
    switch(action.type) {
        case 'ADD-EXPENSE':
            return [...state, action.expenses];
        case 'DELETE-EXPENSE':
            return state.filter((expense) => expense.id != action.id);
            case 'EDIT-EXPENSE':
                return state.map((expense) => {
                    if(expense.id == action.id) {
                        return {
                            ...expense,
                            ...action.updates,
                        }
                    } else {
                        return expense
                    }
                });
        case 'SET-EXPENSE':
            console.log(action.expenses)
            return action.expenses;
        default: 
            return state;
    }
}
export default expensesReducers;
