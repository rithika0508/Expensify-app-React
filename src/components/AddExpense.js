import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';
const AddExpense = (props) => {
    return (
        <div>
            <h2>Add Expense</h2>
            <ExpenseForm onSubmit={(expense) => {
                // props.dispatch(addExpense(expense));
                props.onSubmit(expense);
                props.history.push('/')
            }} />
        </div>
    );

};
const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (expense) => dispatch(addExpense(expense)) 
    }
}
const Connected = connect(undefined,)(AddExpense);
export default Connected;