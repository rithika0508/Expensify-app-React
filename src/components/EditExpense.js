import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
const EditExpense = (props) => {
    return (
        <div>
            This is edit expense page of id {props.match.params.id}
            <ExpenseForm expense={props.expense} onSubmit={(expense) => {
                const id=props.match.params.id;
                props.dispatch(editExpense(id, expense))
                props.history.push('/')
            }}/>
        </div>
    );
};
const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id == props.match.params.id),
        allexpenses: state.expenses
    }
}
const Connected = connect(mapStateToProps)(EditExpense)
export default Connected;