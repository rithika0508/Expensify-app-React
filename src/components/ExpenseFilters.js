import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

class ExpenseFilters extends React.Component {
    state = {
        calendarFocused: null,
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    render() {
        return (
            <div>
                <input
                type="text" 
                defaultValue={this.props.filters.text} 
                onChange={(e) => this.props.dispatch(setTextFilter(e.target.value))}
                autoFocus/>
    
                <select defaultValue="date" onChange={(e) => {
                    if(e.target.value=="date") {
                        this.props.dispatch(sortByDate());
                    } else if(e.target.value=="amount") {
                        this.props.dispatch(sortByAmount());
                    }
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
    
                <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                showClearDates={true}
                onFocusChange={(focusedInput) => this.setState(() => ({ calendarFocused:focusedInput }))}
                numberOfMonths={1}
                isOutsideRange={() => false}/>
            </div>
        );
    }
}
const mapStateToProps = (state) => {
    return {
        filters:state.filter
    }
}
const Connected = connect(mapStateToProps)(ExpenseFilters)
export default (Connected);