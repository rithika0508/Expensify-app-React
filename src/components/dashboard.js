import React from 'react';
import ExpenseFilters from './ExpenseFilters';
import AllExpenses from './Allexpenses';

export const ExpenseDashboardPage = (props) => {
    return (
        <div>
            <ExpenseFilters />
           <AllExpenses />
        </div>
    );
};

export default ExpenseDashboardPage;