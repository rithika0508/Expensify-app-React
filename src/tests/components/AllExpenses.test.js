import React from 'react';
import { shallow } from 'enzyme';
import { AllExpenses } from '../../components/Allexpenses';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';

test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<AllExpenses expenses={expenses}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('should render AllExpenses with empty expenses', () => {
    const wrapper = shallow(<AllExpenses expenses={[]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
})
