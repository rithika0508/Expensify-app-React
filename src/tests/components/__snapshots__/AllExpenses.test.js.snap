// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AllExpenses with empty expenses 1`] = `
<div>
  <p>
    Add an expense to get started!
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <SingleExpense
    amount="500"
    createdAt={0}
    description="Water Bill"
    id="1"
    key="1"
    note=""
  />
  <SingleExpense
    amount={1000}
    createdAt={-345600000}
    description="Rent bill"
    id="2"
    key="2"
    note=""
  />
  <SingleExpense
    amount={800}
    createdAt={345600000}
    description="Gas bill"
    id="3"
    key="3"
    note=""
  />
</div>
`;
